components/tenant-fetcher/cmd/main.go

```diff
@@ -65,29 +87,65 @@ func main() {
 		exitOnError(errors.New("missing handler endpoint or tenant path parameter"), "Error while loading app handler config")
 	}
 
+	transact, closeFunc, err := persistence.Configure(ctx, cfg.Database)
+	exitOnError(err, "Error while establishing the connection to the database")
+
+	authenticatorsConfig, err := authenticator.InitFromEnv(envPrefix)
+	exitOnError(err, "Failed to retrieve authenticators config")
+
+	defer func() {
+		err := closeFunc()
+		exitOnError(err, "Error while closing the connection to the database")
+	}()
+
+	uidSvc := uuid.NewService()
+	converter := tenant.NewConverter()
+	repo := tenant.NewRepository(converter)
+	service := tenant.NewService(repo, transact, uidSvc)
+
 	ctx, err = log.Configure(ctx, &cfg.Log)
 	exitOnError(err, "Failed to configure Logger")
 	logger := log.C(ctx)
 
+	middleware := auth.New(
+		cfg.JwksEndpoint,
+		cfg.IdentityZone,
+		cfg.SubscriptionCallbackScope,
+		extractTrustedIssuersScopePrefixes(authenticatorsConfig),
+		cfg.AllowJWTSigningNone,
+	)
+
+	logger.Infof("JWKS synchronization enabled. Sync period: %v", cfg.JWKSSyncPeriod)
+	periodicExecutor := executor.NewPeriodic(cfg.JWKSSyncPeriod, func(ctx context.Context) {
+		err := middleware.SynchronizeJWKS(ctx)
+		if err != nil {
+			logger.WithError(err).Error("An error has occurred while synchronizing JWKS")
+		}
+	})
+	go periodicExecutor.Run(ctx)
+
 	mainRouter := mux.NewRouter()
 	subrouter := mainRouter.PathPrefix(cfg.RootAPI).Subrouter()
+	subrouter.Use(middleware.Handler())
 
 	logger.Infof("Registering Tenant Onboarding endpoint on %s...", cfg.HandlerEndpoint)
-	subrouter.HandleFunc(cfg.HandlerEndpoint, getOnboardingHandlerFunc(cfg.TenantPathParam)).Methods(http.MethodPut)
+	subrouter.HandleFunc(cfg.HandlerEndpoint, getOnboardingHandlerFunc(service, cfg.TenantPathParam, cfg.TenantProviderAccountId)).Methods(http.MethodPut)
 
 	logger.Infof("Registering Tenant Decommissioning endpoint on %s...", cfg.HandlerEndpoint)
-	subrouter.HandleFunc(cfg.HandlerEndpoint, getDecommissioningHandlerFunc(cfg.TenantPathParam)).Methods(http.MethodDelete)
+	subrouter.HandleFunc(cfg.HandlerEndpoint, getDecommissioningHandlerFunc(service, cfg.TenantPathParam, cfg.TenantProviderAccountId)).Methods(http.MethodDelete)
 
+	healthCheckSubrouter := mainRouter.PathPrefix(cfg.RootAPI).Subrouter()
 	logger.Infof("Registering readiness endpoint...")
-	subrouter.HandleFunc("/readyz", newReadinessHandler())
+	healthCheckSubrouter.HandleFunc("/readyz", newReadinessHandler())
 
 	logger.Infof("Registering liveness endpoint...")
-	subrouter.HandleFunc("/healthz", newReadinessHandler())
+	healthCheckSubrouter.HandleFunc("/healthz", newReadinessHandler())
 
-	runMainSrv, shutdownMainSrv := createServer(ctx, cfg, subrouter, "main")
+	runMainSrv, shutdownMainSrv := createServer(ctx, cfg, mainRouter, "main")
```

Comment:
Now, when we'll be processing real requests, we should also add the correlation and logging middlewares, like [here](https://github.com/kyma-incubator/compass/blob/master/components/director/cmd/director/main.go#L213) in director.